pkg/schema/annotations.go

```diff
@@ -20,36 +19,35 @@ const (
 )
 
 type Annotation interface {
-	NewTypeFromAnn() yamlmeta.Type
+	NewTypeFromAnn() (yamlmeta.Type, error)
 }
 
 type TypeAnnotation struct {
-	any          bool
-	inferredType yamlmeta.Type
-	itemPosition *filepos.Position
+	any  bool
+	item yamlmeta.Node
 }
 
 type NullableAnnotation struct {
-	providedValueType yamlmeta.Type
-	itemPosition      *filepos.Position
+	item yamlmeta.Node
 }
 
 // DefaultAnnotation is a wrapper for a value provided via @schema/default annotation
 type DefaultAnnotation struct {
 	val interface{}
 }
 
-func NewTypeAnnotation(ann template.NodeAnnotation, inferredType yamlmeta.Type, pos *filepos.Position) (*TypeAnnotation, error) {
+// NewTypeAnnotation checks the key-word argument provided via @schema/type annotation, and returns wrapper for the annotated node.
+func NewTypeAnnotation(ann template.NodeAnnotation, node yamlmeta.Node) (*TypeAnnotation, error) {
```

Comment:
```suggestion
func NewTypeAnnotation(ann template.NodeAnnotation, item yamlmeta.Node) (*TypeAnnotation, error) {
```
🤔  this variable plays the exact role as `TypeAnnotation.item`; so either they are _both_ `node` or `item`....?!